api/src/certification/complementary-certification/infrastructure/repositories/complementary-certification-badge-repository.js

```diff
@@ -50,4 +51,25 @@ const findAttachableBadgesByIds = async function ({ ids }) {
   });
 };
 
-export { getAllIdsByTargetProfileId, detachByIds, attach, findAttachableBadgesByIds };
+const getAllWithSameTargetProfile = async function (complementaryCertificationBadgeId) {
+  const complementaryCertificationBadges = await knex('complementary-certification-badges')
+    .select('complementary-certification-badges.*')
+    .join('badges', 'badges.id', '=', 'complementary-certification-badges.badgeId')
+    .where(
+      'badges.targetProfileId',
+      '=',
+      // eslint-disable-next-line knex/avoid-injections
+      knex.raw(`
+      (SELECT "target-profiles".id
+      FROM "complementary-certification-badges"
+      JOIN "badges" on "badges".id = "complementary-certification-badges"."badgeId"
+      JOIN "target-profiles" on "target-profiles"."id" = "badges"."targetProfileId"
+      WHERE "complementary-certification-badges".id = ${complementaryCertificationBadgeId}
+      )
+    `),
```

Comment:
Proposition pour retirer le commentaire eslint, en concordance avec [la documentation knex](https://knexjs.org/guide/raw.html#raw-parameter-binding) : 

```suggestion
    knex.raw(
        `
      (SELECT "target-profiles".id
      FROM "complementary-certification-badges"
      JOIN "badges" on "badges".id = "complementary-certification-badges"."badgeId"
      JOIN "target-profiles" on "target-profiles"."id" = "badges"."targetProfileId"
      WHERE "complementary-certification-badges".id = ?
      )
    `,
        [`${complementaryCertificationBadgeId}`],
      ),
```

Reply:
Ouep, très bien ça sera échappé comme ça.
En l'absence de la connaissance sur comment, on avait accepté le comment en étant sûr que ce n'était pas un input utilisateur.

Reply:
Mais on peut à priori ne pas avoir de `raw` du tout